job_crafter.py

```diff
@@ -38,58 +40,65 @@ def get_device_status(self):
         try:
             return self._device_status
         except:
-            self._device_status = utils.get_connection(**self.options).scheduler.get_device_status(
-                    self.board["device_type"] + "_01"
-                    )
+            conn = utils.get_connection(self.cfg)
+            board = "%s_01" % self.board["device_type"]
+            self._device_status = conn.scheduler.get_device_status(board)
             return self._device_status
 
 # Template handling
     def get_template_from_file(self, file):
-        print("template: using %s" % file)
+        logging.debug("template: using %s" % file)
         self.job_template = self.jinja_env.get_template(file)
 
     def save_job_to_file(self, ext="yaml"):
-        try: os.makedirs(self.options["output_dir"])
-        except: pass
-        file = os.path.join(self.options["output_dir"], self.job["job_name"] + "." + ext)
+        try:
+            os.makedirs(self.cfg['output_dir'])
+        except:
+            pass
+
+        file = os.path.join(self.cfg['output_dir'],
+                            self.job["job_name"] + "." + ext)
+
         with open(file, 'w') as f:
             f.write(self.job_template.render(self.job))
-        print(green("File saved to %s" % file))
+
+        logging.info("==> Job file saved to %s" % file)
 
     def send_to_lava(self):
         try:
             dev = self.get_device_status()
             if dev["status"] == "offline":
-                print(red("Device seems offline, not sending the job"))
+                logging.error("Device is offline, not sending the job")
                 return
         except Exception as e:
-            print(red(repr(e)))
-            print(red("Not sending the job"))
+            logging.error('Not sending the job: %s' % e)
             return
-        print("Sending to LAVA")
+
+        logging.debug("Sending to LAVA")
+
         job_str = self.job_template.render(self.job)
-        ret = utils.get_connection(**self.options).scheduler.submit_job(job_str)
-        try:
-            for r in ret:
-                print(green("Job send (id: %s)" % r))
-                print("Potential working URL: ", "%s/scheduler/job/%s" %
-                        (self.options['web_ui_address'], r))
-        except:
-            print(green("Job send (id: %s)" % ret))
-            print("Potential working URL: ", "%s/scheduler/job/%s" %
-                    (self.options['web_ui_address'], ret))
+        ret = utils.get_connection(self.cfg).scheduler.submit_job(job_str)
+        logging.debug("Job sent (id: %s)" % ret)
+        logging.info("==> Job URL: %s/scheduler/job/%s" %
```

Comment:
This won't work anymore if you submit multinode jobs, since `submit_job` will then return a tuple, containing the list of all the jobs IDs created.

Exemple: `==> Job URL: http://lava.free-electrons.com/scheduler/job/['602673.0', '602673.1']`